### PR TITLE
Add `reduce_[nodal|element]_stack()` in `SingleStackUtils`

### LINKED ISSUE #1175: Add support to call functions (`min`, `max`, `sum`) along stack `up_stack(fun, ...)`
# Description

We'll need to be able to compute things like `max(ϕ)` where `ϕ` is a variable along the vertical stack. So, we should implement some function (`up_stack(fun, ...)`, `down_stack(fun, ...)`) to apply along the vertical stack. It would also be nice to return the `z` location of such `min`/`max`.

# Additional context

Add any other reasons why this feature will help with the goals of CLIMA.

# For CLIMA Developers

- [x] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [x] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

### PR DESCRIPTION
**Pull request by kpamnany**
# Description

And a couple of tests. Closes #1175.

I have

- [X] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [X] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [X] There are no open pull requests for this already
- [X] CLIMA developers with relevant expertise have been assigned to review this submission
- [X] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [X] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: src/Utilities/SingleStackUtils/SingleStackUtils.jl [hunk 1]
```diff
@@ -204,4 +207,58 @@ function get_horizontal_variance(
...
+        var::String;
+        vrange::UnitRange = 1:size(Q, 3),
+    ) where {T, dim, N}
+
+Reduce `var` from `vars` within `Q` over all nodal points in the specified
+`vrange` of elements with `op`. Return a tuple `(result, z)` where `result` is
+the final value returned by `op` and `z` is the index within `vrange` where the
+`result` was determined.
+"""
+function reduce_element_stack(
```
Comment (charleskawczynski) [inline src/Utilities/SingleStackUtils/SingleStackUtils.jl:225]: I think we need this to be `reduce_nodal_stack`, we can probably write two methods and use the same format as `get_vars_from_nodal_stack`/`get_vars_from_element_stack`. Aside from that, the behavior looks good.

Comment (ilopezgp) [reply src/Utilities/SingleStackUtils/SingleStackUtils.jl:225]: Does `op` in the description correspond to `f::Function` in the function handle of the description?

Comment (charleskawczynski) [reply src/Utilities/SingleStackUtils/SingleStackUtils.jl:225]: Yeah, that doc string should be fixed. Good catch.

File: src/Utilities/SingleStackUtils/SingleStackUtils.jl [hunk 2]
```diff
@@ -204,4 +208,108 @@ function get_horizontal_variance(
...
+    op::Function,
+    grid::DiscontinuousSpectralElementGrid{T, dim, N},
+    Q::MPIStateArray,
+    vars::Type,
+    var::String;
+    vrange::UnitRange = 1:size(Q, 3),
+) where {T, dim, N}
+    Nq = N + 1
+    nodal_stacks = [
```
Comment (charleskawczynski) [inline src/Utilities/SingleStackUtils/SingleStackUtils.jl:286]: I think we just need
```julia
    return [
```

This way the user can decide how to parse the nodal results.

**Pull request merged**